### PR TITLE
Add getOptionsFromAttrs

### PR DESCRIPTION
**Pull request by vandangnhathung**
- Use a getOptionsFromAttrs library instead of getOptions.

### CODE REVIEW

File: src/_index.js [hunk 1]
```diff
@@ -40,7 +39,20 @@ class Popup{
...
-        this.options = getOptions(this, this.options);
+        this.id = options.id || this.el.id || DEFAULTS.id;
+
+        this.options = getOptionsFromAttribute(
+            {
+                target: this.el,
+                attributeName: ATTRS.init,
+                defaultOptions: {...this.options, id: this.id},
+                numericValues: ['autoShow'],
+                onIsString: (options) => {
```
Comment (phucbm) [inline src/_index.js]: What's the point of this code?

File: src/_index.js [hunk 2]
```diff
@@ -40,7 +39,20 @@ class Popup{
...
         // get options from JSON init
-        this.options = getOptions(this, this.options);
+        this.id = options.id || this.el.id || DEFAULTS.id;
+
+        this.options = getOptionsFromAttribute(
+            {
+                target: this.el,
+                attributeName: ATTRS.init,
+                defaultOptions: {...this.options, id: this.id},
+                numericValues: ['autoShow'],
```
Comment (phucbm) [inline src/_index.js]: Did you check all the numeric values of the options?

**Review by phucbm — CHANGES_REQUESTED** [review]
Check my comments.

File: src/_index.js (hunk 1 above)
Comment (vandangnhathung) [reply src/_index.js]: I have might made some mistakes. I have just pushed a new code replacing for these old codes. Is it good?

File: src/_index.js (hunk 2 above)
Comment (vandangnhathung) [reply src/_index.js]: Oh, there are one more argument "showingTimes"

File: src/_index.js [hunk 3]
```diff
@@ -27,33 +27,42 @@ class Popup{
...
         this.options = getOptionsFromAttribute(
             {
                 target: this.el,
                 attributeName: ATTRS.init,
-                defaultOptions: {...this.options, id: this.id},
+                defaultOptions: DEFAULTS,
                 numericValues: ['autoShow'],
-                onIsString: (options) => {
+                onIsString: dataAttribute => {
                     // data attribute exist => string
```
Comment (phucbm) [inline src/_index.js]: Please rewrite this commit with the right meaning.

File: src/_index.js [hunk 4]
```diff
@@ -27,33 +27,42 @@ class Popup{
...
+                        idFromAttributeString = dataAttribute;
+                    }
                 }
             });
 
+        if(idFromAttributeString){
+            this.options.id = idFromAttributeString;
+        }
+
+        // get options id from init script
```
Comment (phucbm) [inline src/_index.js]: It's "override with `options` from init by JS".

File: src/_index.js [hunk 5]
```diff
@@ -27,33 +27,42 @@ class Popup{
...
+        this.options = {...this.options, ...options};
+
+        // instance get id
```
Comment (phucbm) [inline src/_index.js]: It's "save id from `options`"

**Review by phucbm — APPROVED** [review]

**Pull request merged**